### PR TITLE
[Feature][Api] Refactor org.apache.dolphinscheduler.api.controller.ProjectController

### LINKED ISSUE #10489: [API] Refactor org.apache.dolphinscheduler.api.controller.ProjectController
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

 Refactor org.apache.dolphinscheduler.api.controller.ProjectController
accroding to  [DSIP-7] [API] Suggest refactor the backend api #10257

### Use case

_No response_

### Related issues

[DSIP-7] [API] Suggest refactor the backend api #10257

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tracehh**
## Purpose of the pull request

Fix #10489

### CODE REVIEW

**Comment by SbloodyS** [comment]
Hi @tracehh , please associate the issue.

**Comment by tracehh** [comment]
see issue: https://github.com/apache/dolphinscheduler/issues/10489

**Comment by SbloodyS** [comment]
> see issue: #10489

to target specific issue from PR, we recommend use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, just desc). It would not only connect issue to PR but also close issue automatically when PR is be closed.

**Comment by tracehh** [comment]
fix: https://github.com/apache/dolphinscheduler/issues/10489

**Comment by tracehh** [comment]
Get it, thanks @SbloodyS

**Comment by tracehh** [comment]
@caishunfeng please review it, thx

**Review by caishunfeng — COMMENTED** [review]
LGTM, but UT is missing, please add some UT of it.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+     */
+    @ApiOperation(value = "queryProjectByCode", notes = "QUERY_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
+    })
+    @GetMapping(value = "/{code}", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(QUERY_PROJECT_DETAILS_BY_CODE_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result queryProjectByCode(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Please avoid using `Result`. Replace with pojo.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+     * @param loginUser       login user
+     * @param projectQueryReq projectQueryReq
+     * @return project list which the login user have permission to see
+     */
+    @ApiOperation(value = "queryProjectListPaging", notes = "QUERY_PROJECT_LIST_PAGING_NOTES")
+    @GetMapping(consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(LOGIN_USER_QUERY_PROJECT_LIST_PAGING_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result queryProjectListPaging(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Same here.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
+    })
+    @GetMapping(value = "/{code}", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(QUERY_PROJECT_DETAILS_BY_CODE_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result queryProjectByCode(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                     @PathVariable("code") long code) {
+        Map<String, Object> result = projectService.queryByCode(loginUser, code);
+        return new ProjectResponse(returnDataList(result));
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: I think we should avoid using `returnDataList` since this is not a clear method.

**Review by SbloodyS — COMMENTED** [review]
I think what you describe in #10489 is  you want to refactor the whole `ProjectController`.

There seems to be a lot missing now.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 3 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: if avoid using returnDataList，maybe refactor service method, it is expensive.  can we call returnDataList in pojo?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 1 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: replace it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 3 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: I think this is necessary, because this is one of the purposes of refactoring. The service module only needs to fine tune the returned data structure without modifying the core logic.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectCreateRequest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    private String projectName;
+
+    @ApiModelProperty(example = "this is a project")
+    private String description;
+
+    public String getProjectName() {
+        return projectName;
+    }
+
+    public void setProjectName(String projectName) {
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectCreateRequest.java]: discuss: can we use lombok.Data to reduce the getter and setter code?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 3 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: rewrite projectV2Service？

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectCreateRequest.java (hunk 4 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectCreateRequest.java:37]: both are ok

Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectCreateRequest.java]: ok. use lombok

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 3 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: > rewrite projectV2Service？

No need. Just optimize the existing return structure and error information.

**Comment by zhongjiajie** [comment]
Hi @tracehh It is seem your UT failed, could you please take a look of the failed tests?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+     */
+    @ApiOperation(value = "delete", notes = "DELETE_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
+    })
+    @DeleteMapping(value = "/{code}", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(DELETE_PROJECT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public BooleanResponse deleteProject(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: It's better rename to `deleteProjectBooleanResponse`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+     * @param loginUser       login user
+     * @param projectQueryReq projectQueryReq
+     * @return project list which the login user have permission to see
+     */
+    @ApiOperation(value = "queryProjectListPaging", notes = "QUERY_PROJECT_LIST_PAGING_NOTES")
+    @GetMapping(consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(LOGIN_USER_QUERY_PROJECT_LIST_PAGING_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public ProjectListPagingResponse queryProjectListPaging(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java:157]: Missing swagger request params example.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser login user
+     * @param userId    user id
+     * @return projects which the user have permission to see, Except for items created by this user
+     */
+    @ApiOperation(value = "queryAuthorizedProject", notes = "QUERY_AUTHORIZED_PROJECT_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "userId", value = "USER_ID", dataType = "Int", example = "100")
+    })
+    @GetMapping(value = "/authed-project", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/authed-project")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser   login user
+     * @param projectCode project code
+     * @return users        who have permission for the specified project
+     */
+    @ApiOperation(value = "queryAuthorizedUser", notes = "QUERY_AUTHORIZED_USER_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "projectCode", value = "PROJECT_CODE", dataType = "Long", example = "100")
+    })
+    @GetMapping(value = "/authed-user", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/authed-user")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+    /**
+     * query authorized and user created project
+     *
+     * @param loginUser login user
+     * @return projects which the user create and authorized
+     */
+    @ApiOperation(value = "queryProjectCreatedAndAuthorizedByUser", notes = "QUERY_AUTHORIZED_AND_USER_CREATED_PROJECT_NOTES")
+    @GetMapping(value = "/created-and-authed", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/created-and-authed")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser login user
+     * @param code      project code
+     * @return project detail information
+     */
+    @ApiOperation(value = "queryProjectByCode", notes = "QUERY_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
+    })
+    @GetMapping(value = "/{code}", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/{code}")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser login user
+     * @param userId    user id
+     * @return the projects which user have not permission to see
+     */
+    @ApiOperation(value = "queryUnauthorizedProject", notes = "QUERY_UNAUTHORIZED_PROJECT_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "userId", value = "USER_ID", dataType = "Int", example = "100")
+    })
+    @GetMapping(value = "/unauth-project", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/unauth-project")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+    /**
+     * query all project list
+     *
+     * @param loginUser login user
+     * @return all project list
+     */
+    @ApiOperation(value = "queryAllProjectList", notes = "QUERY_ALL_PROJECT_LIST_NOTES")
+    @GetMapping(value = "/list", consumes = {"application/json"})
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    @GetMapping(value = "/list")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+    /**
+     * query authorized user
+     *
+     * @param loginUser   login user
+     * @param projectCode project code
+     * @return users        who have permission for the specified project
+     */
+    @ApiOperation(value = "queryAuthorizedUser", notes = "QUERY_AUTHORIZED_USER_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "projectCode", value = "PROJECT_CODE", dataType = "Long", example = "100")
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Missing required.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+    /**
+     * query authorized project
+     *
+     * @param loginUser login user
+     * @param userId    user id
+     * @return projects which the user have permission to see, Except for items created by this user
+     */
+    @ApiOperation(value = "queryAuthorizedProject", notes = "QUERY_AUTHORIZED_PROJECT_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "userId", value = "USER_ID", dataType = "Int", example = "100")
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Missing required.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+    /**
+     * query unauthorized project
+     *
+     * @param loginUser login user
+     * @param userId    user id
+     * @return the projects which user have not permission to see
+     */
+    @ApiOperation(value = "queryUnauthorizedProject", notes = "QUERY_UNAUTHORIZED_PROJECT_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "userId", value = "USER_ID", dataType = "Int", example = "100")
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Missing required.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+    /**
+     * delete project by code
+     *
+     * @param loginUser login user
+     * @param code      project code
+     * @return delete result code
+     */
+    @ApiOperation(value = "delete", notes = "DELETE_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Missing required.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+    /**
+     * query project details by project code
+     *
+     * @param loginUser login user
+     * @param code      project code
+     * @return project detail information
+     */
+    @ApiOperation(value = "queryProjectByCode", notes = "QUERY_PROJECT_BY_ID_NOTES")
+    @ApiImplicitParams({
+        @ApiImplicitParam(name = "code", value = "PROJECT_CODE", dataType = "Long", example = "123456")
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Missing required.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 18]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser login user
+     * @return projects which the user create and authorized
+     */
+    @ApiOperation(value = "queryProjectCreatedAndAuthorizedByUser", notes = "QUERY_AUTHORIZED_AND_USER_CREATED_PROJECT_NOTES")
+    @GetMapping(value = "/created-and-authed", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(QUERY_AUTHORIZED_AND_USER_CREATED_PROJECT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public ProjectListResponse queryProjectCreatedAndAuthorizedByUser(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java:267]: Missing swagger request params example.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 19]
```diff
@@ -0,0 +1,277 @@
...
+     *
+     * @param loginUser login user
+     * @return all project list
+     */
+    @ApiOperation(value = "queryAllProjectList", notes = "QUERY_ALL_PROJECT_LIST_NOTES")
+    @GetMapping(value = "/list", consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(LOGIN_USER_QUERY_PROJECT_LIST_PAGING_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public ProjectListResponse queryAllProjectList(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java:286]: Missing swagger request params example.

**Comment by tracehh** [comment]
@caishunfeng @zhongjiajie please review and if it can merge

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+
+    public List<Project> objToList(Object obj) {
+        List<Project> list = new ArrayList<>();
+        if (obj instanceof List<?>) {
+            for (Object o : (List<?>) obj) {
+                list.add((Project) o);
+            }
+        }
+        return list;
+    }
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java]: It's better to use `JSONUtils` instead of this.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public List<Project> getData() {
+        return data;
+    }
+
+    public void setData(List<Project> data) {
+        this.data = data;
+    }
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java]: Can we use lombok here to avoid getter/setter?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListPagingResponse.java [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public PageInfo<Project> getData() {
+        return data;
+    }
+
+    public void setData(PageInfo<Project> data) {
+        this.data = data;
+    }
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListPagingResponse.java]: Can we use lombok here to avoid getter/setter?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectDeleteBooleanResponse.java [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    public Boolean getData() {
+        return data;
+    }
+
+    public void setData(Boolean data) {
+        this.data = data;
+    }
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectDeleteBooleanResponse.java]: Can we use lombok here to avoid getter/setter?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 24]
```diff
@@ -88,10 +95,11 @@ public class ProjectServiceImpl extends BaseServiceImpl implements ProjectServic
      */
     @Override
     @Transactional
-    public Map<String, Object> createProject(User loginUser, String name, String desc) {
+    public Result createProject(User loginUser, String name, String desc) {
+        Result result = new Result();
 
-        Map<String, Object> result = checkDesc(desc);
-        if (result.get(Constants.STATUS) != Status.SUCCESS) {
+        if (!StringUtils.isEmpty(desc) && desc.length() > 200) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: Why did you change this logic?

Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: instead changing checkDesc(desc);

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java (hunk 20 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java]: ok update it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java (hunk 21 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectListResponse.java]: ok update it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectDeleteBooleanResponse.java (hunk 23 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectDeleteBooleanResponse.java]: ok update it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java (hunk 24 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: I'm not quite sure whether we should do this. PTAL @caishunfeng @ruanwenjun

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 25]
```diff
@@ -541,13 +551,34 @@ private int queryPermission(User user, Project project) {
...
+     * @param loginUser login user
+     * @param project project
+     * @param projectCode project code
+     * @return true if the login user have permission to see the project
+     */
+    @Override
+    public void checkProjectAndAuth(Result result, User loginUser, Project project, long projectCode, String perm) {
+        if (project == null) {
+            putMsg(result, Status.PROJECT_NOT_EXIST);
+        } else if (!canOperatorPermissions(loginUser, new Object[]{project.getId()},AuthorizationType.PROJECTS,perm)) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: Please format this line.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 26]
```diff
@@ -541,13 +551,34 @@ private int queryPermission(User user, Project project) {
...
+     * check project and authorization
+     *
+     * @param result  result
+     * @param loginUser login user
+     * @param project project
+     * @param projectCode project code
+     * @return true if the login user have permission to see the project
+     */
+    @Override
+    public void checkProjectAndAuth(Result result, User loginUser, Project project, long projectCode, String perm) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: Would it be better to use full names instead of abbreviations?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java [hunk 27]
```diff
@@ -206,11 +214,11 @@ public boolean hasProjectAndPerm(User loginUser, Project project, Map<String, Ob
     }
 
     @Override
-    public boolean hasProjectAndPerm(User loginUser, Project project, Result result) {
+    public boolean hasProjectAndPerm(User loginUser, Project project, Result result, String perm) {
         boolean checkResult = false;
         if (project == null) {
             putMsg(result, Status.PROJECT_NOT_FOUND, "");
-        } else if (!canOperatorPermissions(loginUser, new Object[]{project.getId()},AuthorizationType.PROJECTS,PROJECT)) {
+        } else if (!canOperatorPermissions(loginUser, new Object[] {project.getId()}, AuthorizationType.PROJECTS, perm)) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: Please use full name of `perm`

**Comment by tracehh** [comment]
PTAL @caishunfeng @ruanwenjun @zhongjiajie

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java (hunk 24 above)
Comment (caishunfeng) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: I think it's better to keep the `checkDesc` method, because I found some places are repeated judgment logic.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java [hunk 28]
```diff
@@ -401,7 +403,7 @@ public Result querySchedule(User loginUser, long projectCode, long processDefine
         Project project = projectMapper.queryByCode(projectCode);
 
         // check project auth
-        boolean hasProjectAndPerm = projectService.hasProjectAndPerm(loginUser, project, result);
+        boolean hasProjectAndPerm = projectService.hasProjectAndPerm(loginUser, project, result, PROJECT);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java:406]: PTAL @WangJPLeo

**Comment by tracehh** [comment]
ok, update and keep the checkDesc method. @caishunfeng

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java (hunk 28 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java:406]: ok, update and keep the checkDesc method. @caishunfeng

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java (hunk 24 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProjectServiceImpl.java]: ok, update and keep the checkDesc method. @caishunfeng

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java (hunk 28 above)
Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java:406]: PTAL @WangJPLeo

Comment (WangJPLeo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java:406]: No problem here, LGTM.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 29]
```diff
@@ -0,0 +1,288 @@
...
+    @ApiOperation(value = "create", notes = "CREATE_PROJECT_NOTES")
+    @PostMapping(consumes = {"application/json"})
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiException(CREATE_PROJECT_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public ProjectResponse createProject(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                         @RequestBody ProjectCreateRequest projectCreateRequest) {
+        Result result = projectService.createProject(loginUser, projectCreateRequest.getProjectName(),
+            projectCreateRequest.getDescription());
+        return new ProjectResponse(result);
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
        return new createProjectResponse(result);
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectResponse.java [hunk 30]
```diff
@@ -0,0 +1,39 @@
...
+@Data
+public class ProjectResponse extends Result {
+
+    private Project data;
+
+    public ProjectResponse(Result result) {
+        super();
+        this.setCode(result.getCode());
+        this.setMsg(result.getMsg());
+        this.setData((Project) result.getData());
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectResponse.java]: It's better to define a separate entity for each API for subsequent maintenance.

Comment (tracehh) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/project/ProjectResponse.java]: ok

**Review by zhongjiajie — APPROVED** [review]
LGTM, do your guys have other option? @caishunfeng and @SbloodyS

**Review by caishunfeng — APPROVED** [review]
+1

**Review by SbloodyS — APPROVED** [review]
+1
